src/coreclr/jit/targetriscv64.h

```diff
@@ -84,11 +84,11 @@
                                    REG_T0,REG_T1,REG_T2,REG_T3,REG_T4,REG_T5,REG_T6, \
                                    REG_S1,REG_S2,REG_S3,REG_S4,REG_S5,REG_S6,REG_S7,REG_S8,REG_S9,REG_S10,REG_S11
 
-  #define REG_VAR_ORDER_FLT        REG_F12,REG_F13,REG_F14,REG_F15,REG_F16,REG_F17,REG_F18,REG_F19, \
-                                   REG_F2,REG_F3,REG_F4,REG_F5,REG_F6,REG_F7,REG_F8,REG_F9,REG_F10, \
-                                   REG_F20,REG_F21,REG_F22,REG_F23, \
-                                   REG_F24,REG_F25,REG_F26,REG_F27,REG_F28,REG_F29,REG_F30,REG_F31, \
-                                   REG_F1,REG_F0
+  #define REG_VAR_ORDER_FLT        REG_F4, REG_F5, REG_F6, REG_F7, REG_F28, REG_F29, REG_F30, REG_F31, \
+                                   REG_F12, REG_F13, REG_F14, REG_F15, REG_F16, REG_F17, \
+                                   REG_F0, REG_F1, REG_F2, REG_F3, \
+                                   REG_F22, REG_F23, REG_F24, REG_F25, REG_F26, REG_F27, REG_F18, REG_F19, REG_F20, REG_F21, REG_F8, REG_F9, \
+                                   REG_F11, REG_F10
```

Comment:
Could you explain where does this register order come from? In particular:
* the order of temporaries, why ft0-3 are placed after ft4-7, ft8-11, and fa2-7?
* the order of callee-saved, why it's fs6-11, fs3-5, fs0-1?

BTW, small nit: maybe use ABI register names like in REG_VAR_ORDER?

Reply:
I tried to follow LOONGARCH register order according to their types.
Okay. I will update to ABI register names.

Reply:
As far as I know, there is no correct single order for REG_VAR_ORDER and REG_VAR_FLT_ORDER. And I just guess that it tries to avoid frequently used registers in the beginning in ORDER. So registers are mixed according to their types and index in other archs.
If I am wrong. please let me know.

Reply:
That's probably because ARM and ARM64 don't have ABI names for float registers and LA started as a copy from it. But OK, it's just naming, doesn't matter much.

Reply:
> As far as I know, there is no correct single order for REG_VAR_ORDER and REG_VAR_FLT_ORDER. And I just guess that it tries to avoid frequently used registers in the beginning in ORDER. So registers are mixed according to their types and index in other archs.

Yes, there's no correct order. We can figure out a better order for RISC-V when we start optimizing. For a bugfix this will do.